includes/Modules/Analytics/Advanced_Tracking/Measurement_Events/Measurement_Event.php

```diff
@@ -57,15 +57,20 @@ private function validate_config( $config ) {
 			'action'     => false,
 			'selector'   => false,
 			'on'         => false,
+			'metadata'   => false,
 		);
 		foreach ( $config as $key => $value ) {
 			if ( ! array_key_exists( $key, $valid_keys ) ) {
 				throw new \Exception( 'Invalid configuration parameter: ' . $key );
-			}
-			if ( ! is_string( $value ) ) {
-				throw new \Exception( 'Configuration parameter ' . $key . ' must be string' );
-			}
+			}       
 		}
+		if ( ! array_key_exists( 'action', $config ) ) {
+			$config['action'] = null;
+		}
+		if ( ! array_key_exists( 'metadata', $config ) || is_string( $config['metadata'] ) ) {
+			$config['metadata'] = null;
+		}
+		return $config;
```

Comment:
I'm just noticing that the validation here isn't actually strict enough. Right now, it would be possible to specify e.g. `$config = array()` and get away with it. We need to also check that all required parameters are present. Parameters which are optional (e.g. `metadata`) should be filled with defaults that work.

Reply:
This issue has been resolved by checking all required parameters are present in the $config.